src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -195,6 +196,56 @@ public string ReasonPhrase
             }
         }
 
+        public CancellationToken RequestAborted
+        {
+            get
+            {
+                // If a request abort token was previously explicitly set, return it.
+                if (_manuallySetRequestAbortToken.HasValue)
+                {
+                    return _manuallySetRequestAbortToken.Value;
+                }
+                // Otherwise, get the abort CTS.  If we have one, which would mean that someone previously
+                // asked for the RequestAborted token, simply return its token.  If we don't,
+                // check to see whether we've already aborted, in which case just return an
+                // already canceled token.  Finally, force a source into existence if we still
+                // don't have one, and return its token.
+                var cts = _abortedCts;
+                return
+                    cts != null ? cts.Token :
+                    (_requestAborted == 1) ? new CancellationToken(true) :
+                    RequestAbortedSource.Token;
```

Comment:
Ugh. This is copied from Kestrel I know, but once you get to two levels of ternary operation I start to wonder if `if`/`else` would be better :(. Not gonna make you change it though. Rather have the code look the same for now.